backend/branch_relaxation.ml

```diff
@@ -63,11 +63,6 @@ module Make (T : Branch_relaxation_intf.S) = struct
         opt_branch_overflows map pc lbl0 max_branch_offset
           || opt_branch_overflows map pc lbl1 max_branch_offset
           || opt_branch_overflows map pc lbl2 max_branch_offset
-      | Lop (Ipoll { return_label = Some lbl }) ->
-        (* A poll-and-branch instruction can branch to the label lbl,
-           but also to an out-of-line code block. *)
-        code_size + max_out_of_line_code_offset - pc >= max_branch_offset
-        || branch_overflows map pc lbl max_branch_offset
```

Comment:
`Emit` can still generate these shortcuts, but after this PR Linear cannot represent them. We stopped generating the shortcuts when we removed `Linearize` pass, so this is not changing any existing functionality, but the plan was to recover this optimization on the way out of Cfg. I think these shortcuts were added after some benchmarking upstream, and I hestitate to remove them without a trace. On the other hand, leaving a bunch of dead code in `Emit` is not great. What are you thoughts?

Reply:
Actually, `Emit` cannot generate these shortcuts any more after this PR, so I think a comment about it on https://github.com/ocaml-flambda/flambda-backend/issues/2421 would be enough.